commit 6380bf8ff90213282a7a7f99697964a9c2bd6899
Author: Thorsten Blum <thorsten.blum@linux.dev>
Date:   Fri Mar 7 12:35:47 2025 +0100

    module: Replace deprecated strncpy() with strscpy()
    
    strncpy() is deprecated for NUL-terminated destination buffers; use
    strscpy() instead. The destination buffer ownername is only used with
    "%s" format strings and must therefore be NUL-terminated, but not NUL-
    padded.
    
    No functional changes intended.
    
    Link: https://github.com/KSPP/linux/issues/90
    Cc: linux-hardening@vger.kernel.org
    Signed-off-by: Thorsten Blum <thorsten.blum@linux.dev>
    Link: https://lore.kernel.org/r/20250307113546.112237-2-thorsten.blum@linux.dev
    Signed-off-by: Petr Pavlu <petr.pavlu@suse.com>

diff --git a/kernel/module/main.c b/kernel/module/main.c
index 81ecb9d3a935..e1ed5d82f45f 100644
--- a/kernel/module/main.c
+++ b/kernel/module/main.c
@@ -1179,7 +1179,7 @@ static const struct kernel_symbol *resolve_symbol(struct module *mod,
 
 getname:
 	/* We must make copy under the lock if we failed to get ref. */
-	strncpy(ownername, module_name(fsa.owner), MODULE_NAME_LEN);
+	strscpy(ownername, module_name(fsa.owner), MODULE_NAME_LEN);
 unlock:
 	mutex_unlock(&module_mutex);
 	return fsa.sym;